src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -839,11 +854,40 @@ public void SetAbi (string abi)
 			int code = 1;
 			if (!string.IsNullOrEmpty (VersionCode)) {
 				code = Convert.ToInt32 (VersionCode);
-				if (code > 0xffff || code < 0)
-					throw new ArgumentOutOfRangeException ("VersionCode", "VersionCode is outside 0, 65535 interval");
+				if (code > maxVersionCode || code < 0)
+					throw new ArgumentOutOfRangeException ("VersionCode", $"VersionCode is outside 0, {maxVersionCode} interval");
 			}
 			code |= GetAbiCode (abi) << 16;
 			VersionCode = code.ToString ();
 		}
+
+		public void CalculateVersionCode (string currentAbi, string versionCodePattern, string versionCodeProperties)
+		{
+			var regex = new Regex ("\\{(?<key>([A-Za-z]*)):?0*[\\}]");
```

Comment:
After thinking a bit more I think we should only support integer format padding strings.. Since the version code MUST be an integer it makes no sense to support anything else. 

So we should only support

`{0:00000}`
`{0:D4}`

for padding.